app/presenters/debug_presenter.rb

```diff
@@ -0,0 +1,96 @@
+module Presenters
+  class DebugPresenter
+    attr_reader :content_id
+
+    def initialize(content_id)
+      @content_id = content_id
+    end
+
+    def content_items
+      @content_items ||= ContentItem.where(content_id: content_id)
+    end
+
+    def user_facing_versions
+      UserFacingVersion.join_content_items(content_items).pluck("user_facing_versions.number")
+    end
+
+    def locales
+      State.join_content_items(
+        Translation
+        .join_content_items(content_items))
+        .reorder('')
+        .select("locale, content_items.id as id, states.name as state")
+    end
+
+    def latest_content_items
+      ::Queries::GetLatest.call(content_items)
+    end
+
+    def latest_state_with_locale
+      Translation.join_content_items(State.join_content_items(latest_content_items)).pluck("translations.locale, states.name")
+    end
+
+    def link_set
+      @link_set ||= LinkSet.find_by(content_id: content_id)
+    end
+
+    def expanded_links
+      links = ::Queries::GetExpandedLinks.call(content_id, "en")
+      rows = []
+      rows << [links[:content_id], '', 'current']
+      links[:expanded_links].each do |type, sub_links|
+        rows << [type, links[:content_id], 'current']
+        sub_links.each do |link|
+          rows << [
+            {
+              v: (link[:content_id].to_s + type.to_s),
+              f: "<a href='debug?content_id=#{link[:content_id]}'>#{link[:content_id]}</a>"
+            },
+            type,
+            ''
+          ]
+        end
+      end
+
+      rows
+    end
+
+    def states
+      @states = []
+      locales.group_by(&:locale).each do |locale, local_states|
+        @states << [locale, '', '']
+        @states << [{ v: local_states.first.id.to_s, f: local_states.first.state }, locale, '']
+        local_states[1..-1].each_with_index do |state, index|
+          @states << [{ v: state.id.to_s, f: state.state }, local_states[(index + 1) - 1].try(:id).to_s, '']
+        end
+      end
+      @states
+    end
+
+    def event_timeline
+      events.map do |event|
+        [event.action, '', event.id.to_s, event.created_at, event.created_at + 1.second]
```

Comment:
Yeah, annoyingly the empty string are there to map to google charts table definitions. I have removed some of them that I am no longer using to clean it up a bit, but some are still required in that order.